### DISCUSSION TITLE
Enable autonomous collection import

### DISCUSSION THREAD

**Discussion by jgunzelman88** [pocketbase/pocketbase#2734] [Q&A]
I'm trying to create a initialization script and it seems the web client does some manipulation of the schema before submitting it.
It is possible to just have the json from exporting your collection be the same as the one for importing?

**Comment by ganigeorgiev** [comment]
I'm not sure I follow. The export/import json format should be the same.

Could you elaborate what is your use case and what you are trying to do?

**Reply by jgunzelman88** [reply]
I'm trying to jump start a container with the collection schema with no user intervention.
When I export a schema to json I and try to post that schema via curl to api/import it fails.  So I looked at the request that is made from your UI and the import schema is completely different from the json that is exported.  I assume it's to facilitate the comparison that is generated in the UI but I have no way at least to my know to generate the /import payload except through the UI.

**Reply by ganigeorgiev** [reply]
@jgunzelman88 I'm sorry, I still don't understand to what do you refer. Please elaborate or provide an example what do you mean by _"import schema is completely different from the json that is exported"_. Which collection fields are different?

For the available collection web APIs you can check https://pocketbase.io/docs/api-collections/. The "export" from the  Admin UI is actually the collections list api. The import api accept a `collections` field which you can feed with the previous result.

Note that you can also use [Migrations](https://pocketbase.io/docs/migrations/). The prebuilt executable comes with "automigrate" enabled by default, meaning that it will generate a migration file for you in a `pb_migrations` folder with every collection change made in the Admin UI. This folder is safe to be committed in your repo.
Every unapplied migration from the `pb_migrations` folder will be applied automatically on `serve`.
You can also create a snapshot migration with all of your collections configuration by running `./pocketbase migrate collections`
